influxql/functions.go

```diff
@@ -357,6 +357,70 @@ func (r *IntegerMovingAverageReducer) Emit() []FloatPoint {
 	}
 }
 
+// FloatCumulativeSumReducer cumulates the values from each point.
+type FloatCumulativeSumReducer struct {
+	curr    FloatPoint
+	emitted bool
+}
+
+// NewFloatCumulativeSumReducer creates a new FloatCumulativeSumReducer.
+func NewFloatCumulativeSumReducer() *FloatCumulativeSumReducer {
+	return &FloatCumulativeSumReducer{
+		curr: FloatPoint{Nil: true},
+	}
+}
+
+func (r *FloatCumulativeSumReducer) AggregateFloat(p *FloatPoint) {
+	if !r.curr.Nil && r.curr.Time == p.Time {
```

Comment:
All of the other stream processors will skip points at the same time such as `derivative()` and `difference()` so I had this one do it too. It's to avoid having it emit multiple values at the same time. I can remove this though. I don't think I have the same thing with `moving_average()` even though it's also a stream processor. Which do you think is more correct?